gsy_e_sdk/setups/asset_api_scripts/redis_basic_strategies.py

```diff
@@ -0,0 +1,198 @@
+"""
+Template file for a trading strategy through the gsy-e-sdk api client using Redis.
+"""
+
+from time import sleep
+from typing import List, Dict
+from gsy_e_sdk.redis_aggregator import RedisAggregator
+from gsy_e_sdk.clients.redis_asset_client import RedisAssetClient
+
+ORACLE_NAME = "oracle"
+
+# List of assets' names to be connected with the API
+LOAD_NAMES = ["Load 1 L13", "Load 2 L21", "Load 3 L17"]
+PV_NAMES = ["PV 1 (4kW)", "PV 3 (5kW)"]
+STORAGE_NAMES = ["Tesla Powerwall 3"]
+
+# Frequency of bids/offers posting in a market slot - to leave as it is
+TICK_DISPATCH_FREQUENCY_PERCENT = 10
+
+
+class Oracle(RedisAggregator):
+    """Class that defines the behaviour of an "oracle" aggregator."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.is_finished = False
+        self.asset_strategy = {}
+
+    def on_market_cycle(self, market_info):
+        """Place a bid or an offer whenever a new market is created."""
+        if self.is_finished is True:
+            return
+        self.build_strategies(market_info)
+        self.post_bid_offer()
+
+    def on_tick(self, tick_info):
+        """Place a bid or an offer each 10% of the market slot progression."""
+        rate_index = int(float(tick_info["slot_completion"].strip("%")) /
+                         TICK_DISPATCH_FREQUENCY_PERCENT)
+        self.post_bid_offer(rate_index)
+
+    def build_strategies(self, market_info):
+        """
+        Assign a simple strategy to each asset in the form of an array of length 10,
+        ranging between Feed-in Tariff and Market Maker rates.
+        """
+        fit_rate = market_info["feed_in_tariff_rate"]
+        market_maker_rate = market_info["market_maker_rate"]
+        med_price = (market_maker_rate - fit_rate) / 2 + fit_rate
+
+        for area_uuid, area_dict in self.latest_grid_tree_flat.items():
+            if "asset_info" not in area_dict or area_dict["asset_info"] is None:
+                continue
+            self.asset_strategy[area_uuid] = {}
+            self.asset_strategy[area_uuid]["asset_name"] = area_dict["area_name"]
+            self.asset_strategy[area_uuid][
+                "fee_to_market_maker"
+            ] = self.calculate_grid_fee(
+                area_uuid,
+                self.get_uuid_from_area_name("Market Maker"),
+                "current_market_fee",
+            )
```

Comment:
Nit: I'd suggest moving the fee into a variable (since it's being accessed multiple times below) and maybe refactoring the dictionary instantiation:
```suggestion
            fee_to_market_maker = self.calculate_grid_fee(
                area_uuid,
                self.get_uuid_from_area_name("Market Maker"),
                "current_market_fee")
            
            self.asset_strategy[area_uuid] = {
                "asset_name": area_dict["area_name"],
                "fee_to_market_maker": fee_to_market_maker
            }
```